lib/opensearch-config/node-config.ts

```diff
@@ -66,86 +66,114 @@ export enum arm64Ec2InstanceType {
   C6G_LARGE = 'c6g.large',
   C6G_XLARGE = 'c6g.xlarge',
   C6G_2XLARGE = 'c6g.2xlarge',
+  C6GD_XLARGE = 'c6gd.xlarge',
+  C6GD_2XLARGE = 'c6gd.2xlarge',
+  C6GD_4XLARGE = 'c6gd.4xlarge',
+  C6GD_8XLARGE = 'c6gd.8xlarge',
   R6G_LARGE = 'r6g.large',
   R6G_XLARGE = 'r6g.xlarge',
   R6G_2XLARGE = 'r6g.2xlarge',
   R6G_4XLARGE = 'r6g.4xlarge',
   R6G_8XLARGE = 'r6g.8xlarge',
+  R6GD_XLARGE = 'r6gd.xlarge',
+  R6GD_2XLARGE = 'r6gd.2xlarge',
+  R6GD_4XLARGE = 'r6gd.4xlarge',
+  R6GD_8XLARGE = 'r6gd.8xlarge',
   G5G_LARGE = 'g5g.large',
   G5G_XLARGE = 'g5g.xlarge'
 }
+export interface InstanceTypeInfo {
+  instance: InstanceType;
+  hasInternalStoage: boolean;
+}
 
-export const getX64InstanceTypes = (instanceType: string) => {
+export const getX64InstanceTypes = (instanceType: string): InstanceTypeInfo => {
   switch (instanceType) {
   case x64Ec2InstanceType.M5_XLARGE:
-    return InstanceType.of(InstanceClass.M5, InstanceSize.XLARGE);
+    return { instance: InstanceType.of(InstanceClass.M5, InstanceSize.XLARGE), hasInternalStoage: false };
```

Comment:
Typo in all occurrences:
```suggestion
    return { instance: InstanceType.of(InstanceClass.M5, InstanceSize.XLARGE), hasInternalStorage: false };
```